src/test/java/us/kbase/test/narrativemethodstore/db/mongo/MongoDynamicRepoDB2Test.java

```diff
@@ -76,12 +78,13 @@ public static void afterClass() throws Exception {
     
     @Before
     public void before() throws Exception {
-        final MongoClient mc = new MongoClient("localhost:" + MONGO.getMongoPort());
-        final DB db = mc.getDB(DB_NAME);
-        for (final String name: db.getCollectionNames()) {
-            if (!name.startsWith("system.")) {
-                // dropping collection also drops indexes
-                db.getCollection(name).remove(new BasicDBObject());
+        try (MongoClient mc = MongoClients.create("mongodb://localhost:" + MONGO.getMongoPort())) {
+            MongoDatabase db = mc.getDatabase(DB_NAME);
+            for (final String collectionName: db.listCollectionNames()) {
+                if (!collectionName.startsWith("system.")) {
+                    MongoCollection<Document> collection = db.getCollection(collectionName);
+                    collection.deleteMany(new Document()); // Delete all documents without dropping the indexes
```

Comment:
```suggestion
        try (final MongoClient mc = MongoClients.create("mongodb://localhost:" + MONGO.getMongoPort())) {
            final MongoDatabase db = mc.getDatabase(DB_NAME);
            for (final String collectionName: db.listCollectionNames()) {
                if (!collectionName.startsWith("system.")) {
                    final MongoCollection<Document> collection = db.getCollection(collectionName);
                    collection.deleteMany(new Document()); // Delete all documents without dropping the indexes
```

Or inline `collection`

Reply:
👍

Reply:
:+1: